marianas_process_loop_umi_1.8.0/marianas_process_loop_umi.cwl

```diff
@@ -0,0 +1,68 @@
+class: CommandLineTool
+cwlVersion: v1.0
+id: marianas_process_loop_umi_cwl
+baseCommand:
+  - java
+inputs:
+  - id: fastq1
+    type: File?
+    inputBinding:
+      position: 1
+  - id: fastq2
+    type: File?
+    inputBinding:
+      position: 2
+  - id: umi_length
+    type: int?
+    inputBinding:
+      position: 3
+outputs:
+  - id: processed_fastq_1
+    type: File
+    outputBinding:
+      glob: '$(inputs.fastq1.basename.replace(''.fastq'', ''_umi-clipped.fastq''))'
+  - id: processed_fastq_2
+    type: File
+    outputBinding:
+      glob: '$(inputs.fastq2.basename.replace(''.fastq'', ''_umi-clipped.fastq''))'
+  - id: clipping_info
+    type: File
+    outputBinding:
+      glob: info.txt
+  - id: composite_umi_frequencies
+    type: File
+    outputBinding:
+      glob: composite-umi-frequencies.txt
+label: marianas_process_loop_umi.cwl
+arguments:
+  - position: 0
+    prefix: ''
+    separate: false
+    valueFrom: '-server'
+  - position: 0
+    prefix: ''
+    separate: false
+    valueFrom: '-Xms$(parseInt(runtime.ram)/1000 - 2)g'
```

Comment:
Do you want to make this as same as the other tools with expression from memory inputs as unbound arguments or default?

Reply:
```

  - id: memory_per_job
    type: int?
    doc: Memory per job in megabytes
  - id: memory_overhead
    type: int?
    doc: Memory overhead per job in megabytes
```
```

arguments:
  - position: 0
    valueFrom: "${\n  if(inputs.memory_per_job && inputs.memory_overhead) {\n   \n    if(inputs.memory_per_job % 1000 == 0) {\n    \t\n      return \"-Xmx\" + (inputs.memory_per_job/1000).toString() + \"G\"\n    }\n    else {\n      \n      return \"-Xmx\" + Math.floor((inputs.memory_per_job/1000)).toString() + \"G\" \n    }\n  }\n  else if (inputs.memory_per_job && !inputs.memory_overhead){\n    \n    if(inputs.memory_per_job % 1000 == 0) {\n    \t\n      return \"-Xmx\" + (inputs.memory_per_job/1000).toString() + \"G\"\n    }\n    else {\n      \n      return \"-Xmx\" + Math.floor((inputs.memory_per_job/1000)).toString() + \"G\" \n    }\n  }\n  else if(!inputs.memory_per_job && inputs.memory_overhead){\n    \n    return \"-Xmx15G\"\n  }\n  else {\n    \n  \treturn \"-Xmx15G\"\n  }\n}"
  - position: 0
    prefix: '-jar'
    valueFrom: /usr/local/bin/abra2.jar
```
```
requirements:
  - class: ResourceRequirement
    ramMin: "${\r  if(inputs.memory_per_job && inputs.memory_overhead) {\r   \r    return inputs.memory_per_job + inputs.memory_overhead\r  }\r  else if (inputs.memory_per_job && !inputs.memory_overhead){\r    \r   \treturn inputs.memory_per_job + 2000\r  }\r  else if(!inputs.memory_per_job && inputs.memory_overhead){\r    \r    return 15000 + inputs.memory_overhead\r  }\r  else {\r    \r  \treturn 17000 \r  }\r}"
    coresMin: "${\r  if (inputs.number_of_threads) {\r    \r   \treturn inputs.number_of_threads \r  }\r  else {\r    \r    return 4\r  }\r}"
  - class: DockerRequirement
```